### PR TITLE
Host example: show how to create a host with real-life options

### PR DESCRIPTION
**Pull request by hsanjuan**
- AutoRelay
- NAT
- ConnManager
- QUIC
...

This should hopefully help users create hosts that are a little bit more dial-able than the one in the default example, from the get-go. Also shows how to enable extra transports etc.

Please review to double check I am not doing something stupid with the options and with AutoNATService.

### CODE REVIEW

File: libp2p-host/host.go [hunk 1]
```diff
@@ -22,26 +31,88 @@ func main() {
...
+		// it finds it is behind NAT. Use libp2p.Relay(options...) to
+		// enable active relays and more.
+		libp2p.EnableAutoRelay(),
 	)
 	if err != nil {
 		panic(err)
 	}
 
+	// Now let's setup AutoNAT which helps figuring out if we are behind NAT
+	_, err = autonat.NewAutoNATService(ctx, h2,
```
Comment (Stebalien) [inline libp2p-host/host.go]: This is the _server_ side of autonat. Auto-relay automatically creates an autonat client internally.

File: libp2p-host/host.go [hunk 2]
```diff
@@ -22,26 +31,88 @@ func main() {
...
+		for _, addr := range dht.DefaultBootstrapPeers {
+			pi, _ := peer.AddrInfoFromP2pAddr(addr)
+			// We ignore errors as some bootstrap peers may be down
+			// and that is fine.
+			h2.Connect(ctx, *pi)
+		}
+
+		// Now that we have connected to some peers, we can bootstrap the DHT
+		// and discover even more peers.
+		network.  dht.Bootstrap(ctx)
```
Comment (Stebalien) [inline libp2p-host/host.go]: We can skip this now.

File: libp2p-host/host.go [hunk 3]
```diff
@@ -22,26 +31,88 @@ func main() {
...
+		libp2p.ListenAddrStrings(
+			"/ip4/0.0.0.0/tcp/9000",      // regular tcp connections
+			"/ip4/0.0.0.0/udp/9000/quic", // a UDP endpoint for the QUIC transport
+		),
+		// support TLS connections
+		libp2p.Security(libp2ptls.ID, libp2ptls.New),
+		// support secio connections
+		libp2p.Security(secio.ID, secio.New),
+		// support QUIC
+		libp2p.Transport(libp2pquic.NewTransport),
```
Comment (Stebalien) [inline libp2p-host/host.go]: Let's call this out as experimental.

File: libp2p-host/host.go (hunk 2 above)
Comment (hsanjuan) [reply libp2p-host/host.go]: ah. Is bootstrap automatic now from creation?

File: libp2p-host/host.go (hunk 1 above)
Comment (hsanjuan) [reply libp2p-host/host.go]: Made explicit in comment

File: libp2p-host/host.go (hunk 3 above)
Comment (hsanjuan) [reply libp2p-host/host.go]: added

File: libp2p-host/host.go (hunk 2 above)
Comment (Stebalien) [reply libp2p-host/host.go]: Yes. We bootstrap as soon as we start and whenever we create a new connection when we have an empty routing table.

**Pull request merged**